chaco/tests/test_image_plot.py

```diff
@@ -116,62 +104,61 @@ def plot_comparison(input_image, expected_image, **plot_kwargs):
-
-
-@unittest.skipIf(six.PY3, "Bug in the image plotter in python 3. See GH enthought/enable #95.")
-def test_horizontal_bottom_left():
-    # Horizontal orientation with bottom left origin renders a vertically
-    # flipped image.
-    verify_result_image(RGB, IMAGE[::-1], origin='bottom left')
-
-
-@unittest.skipIf(six.PY3, "Bug in the image plotter in python 3. See GH enthought/enable #95.")
-def test_horizontal_top_right():
-    # Horizontal orientation with top right origin renders a horizontally
-    # flipped image.
-    verify_result_image(RGB, IMAGE[:, ::-1], origin='top right')
-
-
-@unittest.skipIf(six.PY3, "Bug in the image plotter in python 3. See GH enthought/enable #95.")
-def test_horizontal_bottom_right():
-    # Horizontal orientation with top right origin renders an image flipped
-    # horizontally and vertically.
-    verify_result_image(RGB, IMAGE[::-1, ::-1], origin='bottom right')
-
-
-@unittest.skipIf(six.PY3, "Bug in the image plotter in python 3. See GH enthought/enable #95.")
-def test_vertical_top_left():
-    # Vertical orientation with top left origin renders transposed image.
-    verify_result_image(RGB, IMAGE.T, origin='top left', orientation='v')
-
-
-@unittest.skipIf(six.PY3, "Bug in the image plotter in python 3. See GH enthought/enable #95.")
-def test_vertical_bottom_left():
-    # Vertical orientation with bottom left origin renders transposed image
-    # that is vertically flipped.
-    verify_result_image(RGB, (IMAGE.T)[::-1],
-                        origin='bottom left', orientation='v')
-
-
-@unittest.skipIf(six.PY3, "Bug in the image plotter in python 3. See GH enthought/enable #95.")
-def test_vertical_top_right():
-    # Vertical orientation with top right origin renders transposed image
-    # that is horizontally flipped.
-    verify_result_image(RGB, (IMAGE.T)[:, ::-1],
-                        origin='top right', orientation='v')
-
-
-@unittest.skipIf(six.PY3, "Bug in the image plotter in python 3. See GH enthought/enable #95.")
-def test_vertical_bottom_right():
-    # Vertical orientation with bottom right origin renders transposed image
-    # that is flipped vertically and horizontally.
-    verify_result_image(RGB, (IMAGE.T)[::-1, ::-1],
-                        origin='bottom right', orientation='v')
-
-
-if __name__ == "__main__":
-    np.testing.run_module_suite()
+class TestResultImage(unittest.TestCase):
+
+    def verify_result_image(self, input_image, expected_image, **plot_kwargs):
+        # These tests were written assuming uint8 inputs.
+        self.assertEqual(input_image.dtype, np.uint8)
+        self.assertEqual(expected_image.dtype, np.uint8)
+        image_result = rendered_image_result(input_image, **plot_kwargs)
+        rms = calculate_rms(image_result, expected_image)
+        self.assertLess(rms, MAX_RMS_ERROR)
+
+    @unittest.skipIf(six.PY3, "Bug in the image plotter in python 3. See GH enthought/enable #95.")
```

Comment:
Is this still an issue?  Enable #95 still seems like it is an issue, but does this code run in to the problem?

Reply:
Not sure, I'll try commenting this out and seeing what happens.

Reply:
This seems to work with the tests enabled. I don't know enough of the underlying issue to be able to figure out what changed, but I suggest we leave off the decorator for now and run the tests, and see how it goes.